source-control/git.md

```diff
@@ -1,33 +1,46 @@
+## Tools
+
+Use a shell/terminal to work with Git commands and not just to rely on [GUI clients](https://git-scm.com/downloads/guis/). If you're working on Windows, [posh-git](https://github.com/dahlbyk/posh-git) is a great PowerShell environment for Git. Another option is to use [Git bash for Windows](http://www.techoism.com/how-to-install-git-bash-on-windows/). On Linux/Mac, install git and simply use your favorite shell/terminal.
+
+## Working with repositories
+
+### Contributing to an existing repository
 
-# Working with repositories
-## Contributing to an existing repository
 When working on an existing project, `git clone` the repository and ensure you understand the team's branch, merge and release strategy (e.g. through the projects [CONTRIBUTING.md file](https://blog.github.com/2012-09-17-contributing-guidelines/)).
 
-## Creating a new repository
+### Creating a new repository
+
 If you create a new repository, agree on your branch, merge and release strategies upfront and document them in your [CONTRIBUTING.md](https://blog.github.com/2012-09-17-contributing-guidelines/). Also lock the default branches and define who can approve and merge Pull Requests into your default branches.
 
-### Required files in default branch for public repositories
+#### Required files in default branch for public repositories
+
 A public repository needs to have the following files in the root directory of the default branch:
-* a [LICENSE](Templates/LICENSE) file 
+
+* a [LICENSE](Templates/LICENSE) file
 * a [README.md](Templates/README.md) file
 * a [CONTRIBUTING.md](Templates/CONTRIBUTING.md) file
 * reference the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/)
 
-To start to contribute by creating your own branch (e.g. `your_alias/feature_name`) and commit early and often - make your commit comments useful to others by including the *WHAT* and *WHY* (instead of the *HOW*). 
+To start to contribute by creating your own branch (e.g. `your_alias/feature_name`) and commit early and often - make your commit comments useful to others by including the *WHAT* and *WHY* (instead of the *HOW*).
+
+## Commit history
 
-# Commit history
 Agree if you want a linear or non-linear commit history. There are pros and cons to both approaches:
-*  Pro linear: [A tidy, linear Git history](https://www.bitsnbites.eu/a-tidy-linear-git-history/)
-*  Con linear: [Why you should stop using Git rebase](https://medium.com/@fredrikmorken/why-you-should-stop-using-git-rebase-5552bee4fed1)
 
-## Approach for non-linear commit history
+* Pro linear: [A tidy, linear Git history](https://www.bitsnbites.eu/a-tidy-linear-git-history/)
+* Con linear: [Why you should stop using Git rebase](https://medium.com/@fredrikmorken/why-you-should-stop-using-git-rebase-5552bee4fed1)
+
+### Approach for non-linear commit history
+
 Merging `topic` into `master`
-```
+
+```bash
```

Comment:
I'd remove bash definition from these blocks, because it doesn't provide any improved formatting and these blocks aren't necessarily bash related regardless of few commands.

Reply:
Fixed